paging/guava/build.gradle

```diff
@@ -32,8 +32,10 @@ dependencies {
     api(project(":paging:paging-common"))
     api(KOTLIN_STDLIB)
     api("androidx.concurrent:concurrent-futures-ktx:1.1.0-alpha01")
```

Comment:
```suggestion
```

Don't need this anymore since we pull in all of guava now

Reply:
Looks like I was too fast to accept this one, we still need this dependency for `ListenableFuture.await()` and `ResolvableFuture`.

Reply:
Do we use `ResolvableFuture`? Are we able to just depend on `KOTLIN_COROUTINES_GUAVA` directly for `ListenableFuture.await()`?

`androidx.concurrent:concurrent-futures-ktx` is meant to be a minimal guava so it's a bit weird to have both.

Reply:
It was used inside `ListenableFutureRemoteMediator`, but since we now depend on full guava we can replace it with `SettableFuture`. The rest can be replaced by `KOTLIN_COROUTINES_GUAVA`, we can even switch it from `api` to `implementation`.